src/Analysis/Ast/Impl/Analyzer/Handlers/AssignmentHandler.cs

```diff
@@ -99,6 +100,10 @@ internal sealed class AssignmentHandler : StatementHandler {
         private bool IsValidAssignment(string name, Location loc) => !Eval.GetInScope(name).IsDeclaredAfter(loc);
 
         private void HandleNameExpression(NameExpression ne, IMember value) {
+            if (value is IPythonEnumerator e) {
+                value = e.Next; // x = enumerate(y)
```

Comment:
This isn't correct either; this needs to set `x` to the value that is the enumerator, not `next()` it.

Reply:
Yeah, made it more common with iterator, so `__next__` would also work.